File: parser/src/test/java/com/aaroncoplan/waterfall/parser/tests/ModuleParsingTests.java
package com.aaroncoplan.waterfall.parser.tests;

import org.junit.Test;

public class ModuleParsingTests {

    @Test
    public void noWhiteSpace(){
        final String code = "module a{}";
        TestUtils.shouldPass(code);
    }

    @Test
    public void bracketOnNewline() {
        final String code = "module a\n{}";
        TestUtils.shouldPass(code);
    }

    @Test
    public void bracketTwoNewlinesAway() {
        final String code = "module a\n\n{}";
        TestUtils.shouldFail(code);
    }

    @Test
    public void newlinesBetweenBrackets() {
        final String[] code = {
          "module a\n{\n\n}",
          "module b{\n\n\n\n}"
        };
        TestUtils.shouldPass(code);
    }

    @Test
    public void testIllegalModuleNames() {
        final String[] code = {
          "module 9startswithnumber {}",
          "module $dollar$ign {}",
          "module A_B_&illegalcharampersand {}"
        };
        TestUtils.shouldFail(code);
    }

    @Test
    public void testSingleLineInModule() {
        final String code = "module a\n{\n int variable2 = 5\n }";
        TestUtils.shouldPass(code);
    }
}

File: parser/src/test/kotlin/com/aaroncoplan/waterfall/parser/tests/ModuleParsingTests.kt
package com.aaroncoplan.waterfall.parser.tests

import org.junit.Test

class ModuleParsingTests {

    @Test
    fun noWhiteSpace() {
        TestUtils.shouldPass("module a{}")
    }

    @Test
    fun bracketOnNewline() {
        TestUtils.shouldPass("module a\n{}")
    }

    @Test
    fun bracketTwoNewlinesAway() {
        TestUtils.shouldFail("module a\n\n{}")
    }

    @Test
    fun newlinesBetweenBrackets() {
        TestUtils.shouldPass(arrayOf(
            "module a\n{\n\n}",
            "module b{\n\n\n\n}"
        ))
    }

    @Test
    fun testIllegalModuleNames() {
        TestUtils.shouldFail(arrayOf(
            "module 9startswithnumber {}",
            "module \$dollar\$ign {}",
            "module A_B_&illegalcharampersand {}"
        ))
    }

    @Test
    fun testSingleLineInModule() {
        TestUtils.shouldPass("module a\n{\n int variable2 = 5\n }")
    }
}
